packages/core/admin/server/services/transfer/token.js

```diff
@@ -0,0 +1,383 @@
+'use strict';
+
+const { map, isArray, omit, uniq, isNil, difference, isEmpty } = require('lodash/fp');
+const crypto = require('crypto');
+
+const {
+  errors: { ValidationError, NotFoundError },
+} = require('@strapi/utils');
+
+const constants = require('../constants');
+
+const TRANSFER_TOKEN_UID = 'admin::transfer-token';
+const TRANSFER_TOKEN_PERMISSION_UID = 'admin::transfer-token-permission';
+
+/**
+ * @typedef TransferToken
+ *
+ * @property {number|string} id
+ * @property {string} name
+ * @property {string} description
+ * @property {string} accessKey
+ * @property {number} lastUsedAt
+ * @property {number} lifespan
+ * @property {number} expiresAt
+ * @property {(number|TransferTokenPermission)[]} permissions
```

Comment:
```suggestion
 * @property {(number[]|TransferTokenPermission[])} permissions
```
There are essentially two types of permissions it could have, but they shouldn't ever be mixed type arrays or something is wrong (this might also need to be updated in API Tokens)

Reply:
Indeed, nice catch!